### PR TITLE
Bug 1867431 - Remove the app flavour from the "What's new in Firefox" string

### PR DESCRIPTION
**Pull request by t-p-white**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1867431

### CODE REVIEW

File: fenix/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1596,7 +1596,10 @@
     back from ETP details (Ex: Tracking content) -->
     <string name="etp_back_button_content_description">Navigate back</string>
     <!-- About page link text to open what's new link -->
-    <string name="about_whats_new">What’s new in %s</string>
+    <string name="about_whats_new" moz:RemovedIn="122" tools:ignore="UnusedResources">What’s new in %s</string>
+    <!-- About page link text to open the latest release notes link. Fenix only has release notes for 'Release' versions, NOT 'Beta' & 'Nightly'.
+         Note: The word "Firefox" should NOT be translated -->
+    <string name="about_whats_new_2">What’s new in Firefox</string>
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1602]: Is there a reason we can't use another placeholder here, like we usually do with app names?
So the placeholder would always be replaced by "Firefox" (with no flavor to it)
We would have to call out what the placeholder will be replaced with at run time in the comment (it's currently missing, must have fallen through the cracks at the time the string landed)

**Review by Delphine — CHANGES_REQUESTED** [review]

File: fenix/app/src/main/res/values/strings.xml (hunk 1 above)
Comment (Delphine) [reply fenix/app/src/main/res/values/strings.xml:1602]: Example: https://transvision.mozfr.org/?recherche=%251%24s+is+produced+by+Mozilla.&repo=android_l10n&sourcelocale=en-US&locale=fr&search_type=strings

Comment (t-p-white) [reply fenix/app/src/main/res/values/strings.xml]: That's a great suggestion @Delphine 🙂

**Comment by t-p-white** [comment]
cc @ewachowiak

**Review by rahulsainani — APPROVED** [review]

**Review by Delphine — APPROVED** [review]

**Pull request merged**